ygen/protogen.go

```diff
@@ -594,3 +668,57 @@ func genListKeyProto(list *yangDirectory, listName string, listPackage string, s
 
 	return km, nil
 }
+
+// enumInProtoUnionField parses an enum that is within a union and returns the generated
+// enumeration that should be included within a protobuf message for it.
+func enumInProtoUnionField(name string, types []*yang.YangType) (map[string]*protoMsgEnum, error) {
+	enums := map[string]*protoMsgEnum{}
+	for _, t := range types {
+		if t.Kind == yang.Yenum && t.Name == "enumeration" {
+			n := fmt.Sprintf("%s", yang.CamelCase(name))
+			enum, err := genProtoEnum(&yang.Entry{
+				Name: n,
+				Type: t,
+			})
+			if err != nil {
+				return nil, err
+			}
+			enums[n] = enum
+		}
+	}
+
+	return enums, nil
+}
+
+// unionFieldToOneOf takes an input name, a yang.Entry containing a field definition and a mappedType
+// containing the proto type that the entry has been mapped to, and returns the list of fields within
+// the oneof that should be created, along with any enumerations that are contained within the union.
+func unionFieldToOneOf(name string, e *yang.Entry, mtype mappedType) ([]protoMsgField, map[string]*protoMsgEnum, error) {
+	enums, err := enumInProtoUnionField(name, e.Type.Type)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	typeNames := []string{}
+	for tn := range mtype.unionTypes {
+		typeNames = append(typeNames, tn)
+	}
+	sort.Strings(typeNames)
+
+	oofs := []protoMsgField{}
+	for _, t := range typeNames {
+		// Split the type name on "." to ensure that we don't have oneof options
```

Comment:
Consider putting this comment and two lines under it into a helper.

Reply:
This seems too specific to have a helper for, since it's only used here?

Reply:
OK. I guess I have a fetish for helpers ;-)